zeebe/broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/steps/MigrationTransitionStep.java

```diff
@@ -56,8 +56,11 @@ public ActorFuture<Void> transitionTo(
             new ClusterContextImpl(context.getPartitionCount()),
             processingState);
     try {
-      dbMigrator.runMigrations();
+      final var migrationsPerformed = dbMigrator.runMigrations();
       zeebeDbContext.getCurrentTransaction().commit();
+      if (migrationsPerformed.migrations() > 0) {
```

Comment:
❓ What if I have no migrations? Did we backport the other fix?

Reply:
No, i forgot that now :see_no_evil: